frontend/src/lib/DeltaParser.ts

```diff
@@ -129,25 +129,25 @@ function handleNewElementMessage(
   })
 }
 
-function handleNewBlockMessage(
+function handleAddBlockMessage(
   reportElement: ReportElement,
   reportId: string,
-  metadata: IForwardMsgMetadata
+  metadata: IForwardMsgMetadata,
+  deltaBlock: Delta
 ): ReportElement {
   MetricsManager.current.incrementDeltaCounter("new block")
 
-  // There's nothing at this node (aka first run), so initialize an empty list.
-  if (!reportElement) {
-    return ImmutableMap({ element: List(), reportId, metadata })
-  }
-
-  // This node was already a list of elements; no need to change anything.
-  if (reportElement.get("element") instanceof List) {
-    return reportElement
+  // This node was already a list of elements. Update everything but the element.
```

Comment:
There's an opportunity to further consolidate this, something like:

`const list= reportElement && reportElement.get("element") instanceof List ? reportElement.get("element") : List()`

and then always

`return ImmutableMap({ element: list, reportId, metadata, deltaBlock })`